src/utils/QCCalculator.cpp

```diff
@@ -149,6 +149,53 @@ class TOPPQCCalculator :
     return error;
   }
 
+//  void calculateSNident (PeptideHit& hit, MSSpectrum<Peak1D>& spec)
+//  {
+    // TODO
+//  }
+
+  float calculateSNmedian (MSSpectrum<Peak1D>& spec, bool norm = true)
+  {
+    float median = 0;
+    float maxi = 0;
+    spec.sortByIntensity();
+    if(spec.size() % 2 == 0)
```

Comment:
@hroest or @jpfeuffer MSSpectrum currently does not provide iterators that iterate only over the m/z (or intensity) entries. This would be in principle possible by providing an adaptor (untested):
struct PositionIterator : public MSSpectrum::Iterator {
    using MSSpectrum::Iterator;
    int operator*() { return MSSpectrum::Iterator::operator*().getMZ(); }
};
This would allow to use the Median function in our Math StatisticsFunction.h